src/app/modules/core/services/beacon-node.service.ts

```diff
@@ -25,16 +36,41 @@ export class BeaconNodeService {
   ) { }
 
   private apiUrl = this.environmenter.env.validatorEndpoint;
-  private beaconNodeEndpoint: string;
-
-  // Observables.
-  status$ = this.http.get<NodeConnectionResponse>(`${this.apiUrl}/health/node_connection`).pipe(
-    tap((res: NodeConnectionResponse) => {
-      this.beaconNodeEndpoint = res.beaconNodeEndpoint + BEACON_API_SUFFIX;
-    }),
-  );
+  beaconState: BeaconState = new BeaconState();
+
+  private get getBeaconEndpoint$(): BehaviorSubject<string> {
+    if (!this.beaconState.beaconNodeEndpoint$.getValue()) {
+      this.http.get<NodeConnectionResponse>(`${this.apiUrl}/health/node_connection`).subscribe(
+          (res: NodeConnectionResponse) => {
+            this.beaconState.beaconNodeEndpoint$.next("http://" + res.beaconNodeEndpoint + BEACON_API_SUFFIX);
+          },
+          (error) => {
+          }
+        );
+    }
+    return this.beaconState.beaconNodeEndpoint$;
+  }
+
+  private updateBalances(): void {
+    this.getBeaconEndpoint$.pipe(
+      filter((beacanNodeEnpoint) => {
+        return beacanNodeEnpoint !== undefined;
+      }),
+      map(beacanNodeEnpoint => {
+        return this.http.get<ValidatorBalances>(`${beacanNodeEnpoint}/validator/balances`)
+          .subscribe((result) => {
+            this.beaconState.balances$.next(result)
```

Comment:
Avoid logic in the subscribe func. Why can't balances just be an observable instead? We can use switch map